ccmbioinfo/osmp
# Fix Docker

server:
- copy over dependency fixes
- remove incorrect licence
- set main to actual production image entrypoint file
- remove unused package scripts
- NODE_ENV=local -> NODE_ENV=development for Express
- rewrite Dockerfile
- move INDEX_38 to MinIO

test-node:
- copy over dependency fixes
- fix package name
- set main to actual production image entrypoint file
- remove unused package scripts
- remove graphql reference in copied eslintrc
- create Dockerfile

workflow:
- also build, lint, check formatting for test-node
- allow lint and formatting jobs to fail if they have never been run before
- fix bind mount paths

local compose:
- switch to more standard /usr/src/app mount paths
- use node:16-slim
- use init if running yarn script
- don't unnecessarily use entrypoint
- automatically interpolate MONGO_CONNECTION_STRING
- remove extra bridge network definition
- update Keycloak

staging compose:
- consistent restart and logging configuration
- use :latest tag instead of overwriting :dev tag
- use test-node image, don't mount source code
- update keycloak
- PROJECT_ROOT to help workflow fix bind mount paths
